drift-transport-netty/src/main/java/com/facebook/drift/transport/netty/codec/ZlibTransformer.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2013 Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.drift.transport.netty.codec;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufInputStream;
+import io.netty.buffer.ByteBufOutputStream;
+import io.netty.buffer.Unpooled;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.zip.DeflaterOutputStream;
+import java.util.zip.InflaterInputStream;
+
+public class ZlibTransformer
+        implements ByteBufTransformer
+{
+    private static final int ZLIB_COMPRESSION_BUFFER_SIZE = 512;
```

Comment:
`INITIAL_OUTPUT_BUFFER_SIZE_IN_BYTES`

Reply:
Actually I'm not sure if that supposed to be a constant. I would try to chose a better buffer size based on the input. (e.g.: if the compressed input is known to be 1kb already, why do we want to expand the buffer twice?)

Reply:
> Also, we want to try not to unzip the whole message all at once.

Currently it will always unzip the whole message, as the Inflating input stream is used. It fails if the input stream is incomplete.